plugins/modules/inventory_workflow_manager.py

```diff
@@ -1659,134 +1671,426 @@ def handle_partially_provisioned(self, provision_count, device_type):
+            in summarizing the provisioning operation for reporting purposes.
+        """
+
+        if already_provisioned_count == total_devices:
             self.handle_all_already_provisioned(device_ip_list, device_type)
-        elif provision_count == total_devices_to_provisioned:
+        elif provision_count == total_devices:
             self.handle_all_provisioned(device_type)
         elif provision_count == 0:
             self.handle_all_failed_provision(device_type)
         else:
             self.handle_partially_provisioned(provision_count, device_type)
 
-        return self
+    def assign_device_to_site(self, device_ip, assign_network_device_to_site):
+        """
+        Assign a network device to a specified site in Cisco Catalyst Center.
+
+        Parameters:
+            device_ip (str): The IP address of the device to be assigned to the site.
+            assign_network_device_to_site (dict): A dictionary containing the necessary parameters to assign
+                the device to a site, including the device IDs and site ID.
+        Returns:
+            None
+        Description:
+            This function assigns a network device to a site using the Cisco DNA Center API. It logs the process
+            and monitors the assignment task until it completes. If the assignment is successful, it logs a
+            success message; otherwise, it logs an error message.
+
+            The function handles any exceptions that may occur during the API call or while checking the task status
+            to ensure that the application can manage errors gracefully.
+        """
+        time.sleep(10)
+        try:
+            self.log("Assigning device with IP {0} to site. Request data: {1}".format(device_ip, assign_network_device_to_site))
+            response = self.dnac._exec(
+                family="site_design",
+                function='assign_network_devices_to_a_site',
+                op_modifies=True,
+                params=assign_network_device_to_site,
+            )
+            self.log("Received API response from 'assign_network_devices_to_a_site': {0}".format(str(response)), "DEBUG")
+            task_id = response.get("response").get("taskId")
+            if not task_id:
+                self.log("No taskId found in the response for device IP {0}".format(device_ip), "ERROR")
+                return False
+
+            self.log("Task ID for assigning device to site: {0}".format(task_id), "DEBUG")
+            retry_limit = 25.
+            retry_count = 0.
+            while retry_count < retry_limit:
+                execution_details = self.get_task_details(task_id)
+                progress = execution_details.get("progress", "")
+                is_error = execution_details.get("isError", False)
+
+                if "success" in progress:
+                    self.log("Successfully assigned site to the device {0}".format(device_ip))
+                    return True
+                elif is_error:
```

Comment:
It should be if only as you are returning from above 'if' when you meet the required check.

Reply:
can't understand

Reply:
updated the code